source/Calamari.AzureScripting/AzureContextScriptWrapper.cs

```diff
@@ -62,17 +63,28 @@ public AzureContextScriptWrapper(IVariables variables, ICalamariFileSystem fileS
             using (new TemporaryFile(Path.Combine(workingDirectory, "AzureProfile.json")))
             using (var contextScriptFile = new TemporaryFile(CreateContextScriptFile(workingDirectory, scriptSyntax)))
             {
-                if (variables.Get(SpecialVariables.Account.AccountType) == "AzureServicePrincipal")
+                if (variables.Get(SpecialVariables.Account.AccountType) == "AzureServicePrincipal" || variables.Get(SpecialVariables.Account.AccountType) == "AzureOidc")
                 {
-                    SetOutputVariable("OctopusUseServicePrincipal", bool.TrueString);
+                    SetOutputVariable("OctopusAzSpOrOidc", bool.TrueString);
                     SetOutputVariable("OctopusAzureADTenantId", variables.Get(SpecialVariables.Action.Azure.TenantId)!);
                     SetOutputVariable("OctopusAzureADClientId", variables.Get(SpecialVariables.Action.Azure.ClientId)!);
-                    variables.Set("OctopusAzureADPassword", variables.Get(SpecialVariables.Action.Azure.Password));
+
+                    if (variables.Get(SpecialVariables.Account.AccountType) == "AzureServicePrincipal")
+                    {
+                        variables.Set("OctopusUseServicePrincipal", bool.TrueString);
+                        variables.Set("OctopusAzureADPassword", variables.Get(SpecialVariables.Action.Azure.Password));
+                    }
+                    else
+                    {
+                        variables.Set("OctopusOpenIdJwt", variables.Get(AccountVariables.Jwt));
+                    }
```

Comment:
The variables being set here are a bit confusing. For an OIDC account we get `OctopusAzSpOrOidc` set to `true`, but for a password based service principal account we get `OctopusAzSpOrOidc` and `OctopusUseServicePrincipal` set to `true`.
Don't think it is worth changing, but food for thought. Also, not a massive fan of the shortened `OctopusAzSpOrOidc`, would prefer to spell it all out, `OctopusAzureServicePrincipalOrOidc` 🤷

Reply:
I'd meant to tidy this up, I think the scripts mostly had this removed but missed the wrapper. Thanks for the call out